src/Native/CMakeLists.txt

```diff
@@ -26,10 +26,19 @@ endif ()
 
 if (CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64 OR CMAKE_SYSTEM_PROCESSOR STREQUAL amd64)
     add_definitions(-DBIT64=1)
-    add_definitions(-DX64=1)
+    add_definitions(-D_AMD64_)
+elseif (CMAKE_SYSTEM_PROCESSOR STREQUAL x86)
```

Comment:
There was definitely something unexcepted here. I believe cmake was automatically defining X86. But clearly BIT32 was not set for x86.